src/ManageCourses.Api/Services/TransitionService.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Linq;
+using GovUk.Education.ManageCourses.Domain.DatabaseAccess;
+
+namespace GovUk.Education.ManageCourses.Api.Services
+{
+    public class TransitionService : ITransitionService
+    {
+        private IManageCoursesDbContext _context;
+
+        public TransitionService(IManageCoursesDbContext context)
+        {
+            _context = context;
+        }
+
+        public void UpdateNewCourse(string providerCode, string courseCode, string email)
+        {
+            var optedInNewCourse = _context.GetCourse(providerCode, courseCode, email)
+                .FirstOrDefault(c =>
+                    c.Provider.OptedIn &&
+                    c.CourseSites != null &&
+                    c.CourseSites.All(cs => cs.Status.Equals("n", StringComparison.InvariantCultureIgnoreCase)));
+
+            if (optedInNewCourse != null)
+            {
+                foreach(var cs in optedInNewCourse.CourseSites)
+                {
+                    cs.Publish = "Y";
+                    cs.Status = "R";
+                }
+
+                _context.Save();
```

Comment:
Just to check, seeing as I'm not familiar with .NET Core, this save here will update the context but not the DB, is that right? So this change won't be saved to the DB until `_searchAndCompareService.SaveCourse(providerCode, courseCode, name);` finishes in the `PublishController`?

I'm just double-checking that if an enrichment validation fails, or any other validation really, this update won't actually be persisted to the DB.